ush/python/pygfs/task/snow_analysis.py

```diff
@@ -86,6 +87,82 @@ def __init__(self, config: Dict[str, Any]):
+
+        Returns
+        ----------
+        None
+        """
+
+        # create a temporary dict of all keys needed in this method
+        localconf = AttrDict()
+        keys = ['DATA', 'PARMgfs', 'COMIN_OBS', 'OPREFIX', 'snow_prepobs_path']
+        for key in keys:
+            localconf[key] = self.task_config[key]
+
+        # Read and render the prep_snocvr_snomad.yaml.j2
+        logger.info(f"Reading {self.task_config.PREP_SNOCVR_SNOMAD_YAML}")
+        prep_snocvr_snomad_config = parse_j2yaml(self.task_config.PREP_SNOCVR_SNOMAD_YAML, localconf)
+        logger.debug(f"{self.task_config.PREP_SNOCVR_SNOMAD_YAML}:\n{pformat(prep_snocvr_snomad_config)}")
+
+        # define these locations in gdas/snow/prep/prep_ghcn.yaml.j2
+        logger.info("Copying SNOCVR and SNOMAD obs to DATA")
+        FileHandler(prep_snocvr_snomad_config.stage).sync()
+
+        # Execute obsBuilder to create the combined snocvr and snomad in IODA format
+        logger.info("Create the combined snocvr and snomad data in IODA format")
+
+        input_snocvr = f'{localconf.OPREFIX}snocvr.tm00.bufr_d'
+        input_snomad = f'{localconf.OPREFIX}snomad.tm00.bufr_d'
+        output_file = f'{localconf.OPREFIX}snocvr_snomad.tm00.nc'
+        if os.path.exists(f"{os.path.join(localconf.DATA, output_file)}"):
+            rm_p(output_file)
+
+        logger.info("Link OBSBUILDER into DATA/")
+        exe_src = self.task_config.OBSBUILDER
+        exe_dest = os.path.join(self.task_config.DATA, os.path.basename(exe_src))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(exe_src, exe_dest)
+
+        exe = Executable(exe_dest)
+        if os.path.exists(input_snocvr):
+            exe.add_default_arg(["--input_snocvr", f"{os.path.join(localconf.DATA, input_snocvr)}"])
+        exe.add_default_arg(["--output", f"{os.path.join(localconf.DATA, output_file)}"])
+        if os.path.exists(input_snomad):
+            exe.add_default_arg(["--input_snomad", f"{os.path.join(localconf.DATA, input_snomad)}"])
+        try:
+            logger.debug(f"Executing {exe}")
+            exe()
+        except OSError:
+            logger.exception(f"Failed to execute {exe}")
+            raise
+        except Exception as err:
+            logger.exception(f"An error occured during execution of {exe}")
+            raise WorkflowException(f"An error occured during execution of {exe}") from err
+
+        # Ensure the IODA snow depth GHCN file is produced by the IODA converter
+        # If so, copy to DATA/obs/
+        if not os.path.isfile(f"{os.path.join(localconf.DATA, output_file)}"):
+            logger.exception(f"{self.task_config.OBSBUILDER} failed to produce {output_file}")
+            raise FileNotFoundError(f"{os.path.join(localconf.DATA, output_file)}")
+        else:
+            logger.info(f"Copy {output_file} successfully generated")
+            FileHandler(prep_snocvr_snomad_config.netcdf).sync()
```

Comment:
I'm wondering if we should allow the analysis to still run even if this file fails to be produced? Thoughts?